loading.bs

```diff
@@ -0,0 +1,949 @@
+    [=exchange signature/date=], return a failure.
+1. Set |result|'s [=exchange signature/certificate chain=] to the result of
+    [=handling the certificate reference=] |certUrl| with a hash of |result|'s
+    [=exchange signature/certSha256=] in the context of |client|. If this is a
+    failure, return it.
+1. Return |result|.
+
+<h4 algorithm id="handling-cert-url">Handling the certificate reference</h4>
+
+<dfn>Handling the certificate reference</dfn> |certUrl| with the SHA-256 hash
+|certSha256|, in the context of an [=environment settings object=] |client|,
+returns the result of the following steps:
+
+1. Assert: This algorithm is running [=in parallel=].
+1. Let |certRequest| be a new [=request=] with the following items:
+
+    : [=request/url=]
+    :: |certUrl|
+    : [=request/header list=]
+    :: «`` `Accept` ``: `` `application/cert-chain+cbor` ``»
+    : [=request/client=]
+    :: |client|
+    : [=request/service-workers mode=]
+    :: "`none`"
+    : [=request/mode=]
+    :: "`cors`"
+1. Let |certResponse| be the result of [=fetching=] |certRequest|.
+1. If |certResponse|'s [=response/status=] is not `200`, or the [=supplied MIME
+    type=] of |certResponse| is not `"application/cert-chain+cbor"`, return a
+    failure.
+1. If |certResponse|'s [=response/body=] is null or that body's [=body/stream=]
+    is null, return a failure.
+1. Let |bytes| be the result of [=ReadableStream/read all bytes|reading all
+    bytes=] from |certResponse|'s [=response/body=]'s [=body/stream=] with a
+    [=ReadableStream/get a reader|new reader=] over the same stream.
+1. Wait for |bytes| to settle.
+1. If |bytes| was rejected, return a failure.
+1. Let |chain| be the [=certificate chain=] produced by parsing |bytes|' value
+    using the [=cert-chain CDDL=]. If |bytes|'s value doesn't match this CDDL or
+    isn't [=canonically-encoded CBOR=], return a failure.
+1. Assert: |chain| has at least one [=list/item=].
+1. If the [=SHA-256=] hash of |chain|'s [=certificate chain/leaf=]'s [=augmented
+    certificate/certificate=] is not equal to |certSha256|, return a failure.
+1. Return |chain|.
+
+<h3 algorithm id="signed-message">The signed message</h3>
+
+The <dfn>b1 signed message</dfn> for an [=exchange signature=] |signature| and a
+[=byte sequence=] |headerBytes| is the concatenation of the following [=byte
+sequences=]:
+
+1. The byte 0x20 (SP) repeated 64 times. This matches the TLS 1.3 ([[RFC8446]])
+    format to avoid cross- protocol attacks if anyone uses the same key in a TLS
+    certificate and an exchange-signing certificate.
+1. `` `HTTP Exchange 1 b1` `` to serve as a context string.
+
+    Note: Each draft of
+    [[draft-yasskin-httpbis-origin-signed-exchanges-impl-01]] and the final RFC
+    for [[draft-yasskin-http-origin-signed-responses]] will use distinct context
+    strings.
+1. A single 0x00 byte which serves as a separator.
+1. The concatenation of the following [=byte sequences=], which is the
```

Comment:
@davidben How do you feel about this message algorithm? I think we can simplify it for b2, since I only made it this complicated to avoid confusion between multiple versions of the message, and that's now handled by the `HTTP Exchange 1 b1` context string, but it'll inform whether we can go to Origin Trial with b1 or have to go to b2 first.